DataSourceServices.cs

```diff
@@ -223,18 +223,35 @@ public sealed class DataSourceServices:IDataSource {
         /// </summary>
         /// <returns>Instance of TwIdentity.</returns>
         private TwIdentity _GetDSIdentity() {
-            var _asm=this.HandlerType.Assembly;
-            var _ds=_asm.GetCustomAttributes(typeof(DataSourceAttribute), false)[0] as DataSourceAttribute;
-            var _groups=(SupportedGroupsAttribute)Attribute.GetCustomAttribute(this.HandlerType, typeof(SupportedGroupsAttribute),true)??new SupportedGroupsAttribute(TwDG.DS2);
-            var _version=new Version(((AssemblyFileVersionAttribute)Attribute.GetCustomAttribute(_asm, typeof(AssemblyFileVersionAttribute), false)??new AssemblyFileVersionAttribute("1.0.0.0")).Version);
-
+            var asm = HandlerType.Assembly;
```

Comment:
Code formatting should be the same as in the whole project (including brackets, variable names, indents, etc.)
Refactoring existing variables unacceptable.
Using the "this" keyword for instance items (as field, properties and methods) is mandatory